template-for-recording/record_screen_and_upload.sh

```diff
@@ -180,33 +180,38 @@ PARAM_SOURCECODE_DIR="--sourcecode ${APP_HOME}"
 #   java version "12-ea"            "12-ea
 JAVA_VERSION=$($JAVACMD -version 2>&1 | grep '"9\|"10\|"11\|"12')
 
-# Only include digits from value in JAVA_VERSION, 
-# If the above value contains non-numeric values, 
+# Only include digits from value in JAVA_VERSION,
+# If the above value contains non-numeric values,
 # only the digits before it are returned
 #
 # For e.g.
-#   JAVA_VERSION   JAVA_VERSION_INT_VALUE   
+#   JAVA_VERSION   JAVA_VERSION_INT_VALUE
 #   ============   ======================
 #    <no output>         <no output>
 #         9                  9
 #        10                  10
 #        11                  11
 #       12-ea                12
 #
-# The for loop below iterates through tokens, 
+# The for loop below iterates through tokens,
 # tokens are separated by the delimeter provided to 'delim'
-# Only the first token is retained, and as per the above table, 
+# Only the first token is retained, and as per the above table,
 # it will always be empty or a numeric value between 9 and 12 (included)
 JAVA_VERSION_INT_VALUE=$(echo $JAVA_VERSION | grep -o '"[0-9]*' | tr -d '"')
 if [ -z "${JAVA_VERSION_INT_VALUE}" ]; then
```

Comment:
It would be nice to use the same conditional logic as the one we have in the Windows script

Reply:
I know it not ideal.

Unfortunately, the way we do it in Windows, also because of the batch language limitations tokens are used to break the version string, while in bash we use grep.

We could achieve it but would it be worth the time and effort.